core/Symbols.cc

```diff
@@ -60,8 +61,9 @@ TypePtr Symbol::externalType(const GlobalState &gs) const {
         } else {
             vector<TypePtr> targs;
             for (auto tm : typeMembers()) {
-                if (tm.data(gs)->isFixed()) {
-                    targs.emplace_back(tm.data(gs)->resultType);
+                auto *lambdaParam = cast_type<LambdaParam>(tm.data(gs)->resultType.get());
+                if (lambdaParam && lambdaParam->isFixed()) {
+                    targs.emplace_back(lambdaParam->lower);
```

Comment:
ditto

Reply:
Again, `lower` and `upper` will be the same type, if the type variable is marked as being fixed.